### PR TITLE
Use paging for fetching schema tables from the cluster

### PR DESCRIPTION
**Pull request by haaawk**
Those queries can sometimes return large number of rows
and non-paged can put unnecessary stress on the cluster.

### CODE REVIEW

File: driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import static com.datastax.driver.core.SchemaElement.TYPE;
```
Comment (avelanarius) [inline driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java]: Please add "Modified by" at the top, at line 16 (not needed in other file, as it already has this notice from before):

```
/*
 * Copyright (C) 2021 ScyllaDB
 *
 * Modified by ScyllaDB
 */
```

Comment (haaawk) [reply driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java]: done

**Review by avelanarius — APPROVED** [review]
LGTM

**Comment by tarzanek** [comment]
I verified and tested this change
all system_schema queries now use 10000 pageing limit - the big question is, if we want this configurable, Vlad thinks it would make sense

List of queries coming from driver after this change that still don't use any page (the peers_v2 select is weird, I guess we should get some support for this table on our side - new bug?) : 

SELECT * FROM system.local WHERE key="local"
SELECT * FROM system.peers_v2
SELECT * FROM system.peers
select cluster_name from system.local where key = "local"
USE "mykeyspace"

also note that 
select cluster_name from system.local where key = "local"
was called A LOT for having just a single session (scylla side was 4.4.3) - a bug?

I can share the session trace, or you can just repeat it with enabled 100% tracing and running a sample java app with e.g. select/insert with this driver

**Comment by tarzanek** [comment]
attaching sample scylla side tracing logs, select, insert, select, delete, select is the sample app (at the bottom)
single node only
[tracing-java-driver.ods](https://github.com/scylladb/java-driver/files/7235835/tracing-java-driver.ods)

**Review by tarzanek — APPROVED** [review]
Unless we want to make the 10000 (which is double the default 5000 size) parametrized, I am OK.

Or even better - why don't we use the default paging limit here?

**Comment by haaawk** [comment]
@Lorak-mmk has checked that this PR actually does not work because the queries are done before session is created. @Lorak-mmk could you share your findings here please.

**Comment by Lorak-mmk** [comment]
When there is a lot of tables in Scylla, the results actually take more than one page. At the `ArrayBackedResultSet.java:122` we can see:
```
        // info can be null only for internal calls, but we don't page those. We assert
        // this explicitly because MultiPage implementation doesn't support info == null.
        assert r.metadata.pagingState == null || info != null;

        return r.metadata.pagingState == null
            ? new SinglePage(
                columnDefs, tokenFactory, protocolVersion, columnDefs.codecRegistry, r.data, info)
            : new MultiPage(
                columnDefs,
                tokenFactory,
                protocolVersion,
                columnDefs.codecRegistry,
                r.data,
                info,
                r.metadata.pagingState,
                session);
```
however, now we performed multi-page query in the session initialization - so the paging state is not null, but info (type: `ExecutionInfo`) is null. If asserts were enabled, it would fail. As they are not by default, the code continues and throws NPE later.

`MultiPage` does in fact use `info`, I don't think we can stop requiring it. I tried to put together some `ExecutionInfo` so that maybe it works.

```
private static ResultSetFuture queryAsync(
      Cluster cluster, String query, Connection connection, ProtocolVersion protocolVersion)
      throws ConnectionException, BusyConnectionException {
    ExecutionInfo info =
        new ExecutionInfo(
                0,
                0,
                ImmutableList.of(cluster.manager.metadata.getHost(connection.endPoint)),
                null,
                null)
            .with(null, null, null, new SimpleStatement(query), protocolVersion, null);
    DefaultResultSetFuture future =
        new DefaultResultSetFuture(
            null,
            protocolVersion,
            new Requests.Query(query, QueryProtocolOptions.DEFAULT.copy(10000), false));
    future.setDefaultExecutionInfo(info);
    connection.write(future);
    return future;
  }
```
(`setDefaultExecutionInfo`  is just a small method that I added to put this info in future, it will overwrite `info` if `pagingState` is not null, and `info` is null). When this code is run, there is another problem - `MultiPage` also needs `SessionManager`, which we don't have in `queryAsync` method, or `SchemaParser` at all - and I don't know how to get it. Someone who knows this codebase better should take a look.

**Pull request closed without merging**